### PR TITLE
cephadm: also return JSON decode error

### PR DESCRIPTION
**Pull request by sebastian-philipp**
This PR should also help with https://tracker.ceph.com/issues/44209

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -3394,10 +3394,14 @@ if __name__ == "__main__":
     try:
         r = args.func()
     except Error as e:
+        logger.exception('execution failed')
         if args.verbose:
             raise
         sys.stderr.write('ERROR: %s\n' % e)
         sys.exit(1)
+    except Exception:
```
Comment (mgfritch) [inline src/cephadm/cephadm]: catch everything?

```suggestion
    except:
```

File: src/cephadm/cephadm [hunk 2]
```diff
@@ -3394,10 +3394,14 @@ if __name__ == "__main__":
     try:
         r = args.func()
     except Error as e:
+        logger.exception('execution failed')
```
Comment (mgfritch) [inline src/cephadm/cephadm]: This adds a traceback to the console in all `Error` cases, which is what I though we were trying to avoid ... ?

Comment (sebastian-philipp) [reply src/cephadm/cephadm]: right. tried to add this just to the log file, not the console

Comment (sebastian-philipp) [reply src/cephadm/cephadm]: Do we need two loggers? one console+file logger and a dedicated console-only logger?

**Comment by sebastian-philipp** [comment]
QA run failed: https://tracker.ceph.com/issues/44302

**Pull request merged**